hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadata.java

```diff
@@ -106,5 +109,13 @@ static HoodieTableMetadata create(HoodieEngineContext engineContext, HoodieMetad
    */
   Option<String> getSyncedInstantTime();
 
+  /**
+   * Get the instant time to which the metadata is synced w.r.t data timeline on the reader side.
+   *
+   * This is different from the getSyncedInstantTime() because the reader should sync all completed instants
+   * from the data timeline in order to always provide the most up to date view of the files within the dataset.
+   */
+  Option<String> getSyncedInstantTimeForReader();
```

Comment:
So we need this only for tests? 

```
./hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadata.java:  Option<String> getSyncedInstantTimeForReader();
./hudi-common/src/main/java/org/apache/hudi/metadata/BaseTableMetadata.java:  public Option<String> getSyncedInstantTimeForReader() {
./hudi-common/src/main/java/org/apache/hudi/metadata/FileSystemBackedTableMetadata.java:  public Option<String> getSyncedInstantTimeForReader() {
./hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/functional/TestHoodieBackedMetadata.java:      assertEquals(metadata.getSyncedInstantTimeForReader().get(), newCommitTime);
```

Reply:
Yes

Reply:
Can we avoid this if we can. This is a pretty important interface, having additional methods for testing may be an overkill.?
Happy to take this up

Reply:
Removed from public interface. Can easily be accessed in testing by casting the interface.